website/docs/guides/errors.md

```diff
@@ -0,0 +1,190 @@
+  throw new Error('boom');
+}
+
+main(function*() {
+  try {
+    yield tickingBomb()
+  } catch(err) {
+    console.log("it blew up:", err.message);
+  }
+});
+```
+
+However, note that something interesting happens when we instead `spawn` the
+`tickingBomb` operation:
+
+``` typescript
+import { main } from 'effection';
+import { tickingBomb } from './ticking-bomb';
+
+main(function*() {
+  yield spawn(tickingBomb());
+  try {
+    yield; // sleep forever
+  } catch(err) {
+    console.log("it blew up:", err.message);
+  }
+});
+```
+
+You might be surprised that we do *not* enter the catch handler here! Instead,
+our entire main task just fails. This is by design! We are only allowed to
+catch errors thrown by whatever we yield to directly, not by any spawned
+children or resources running in the background. This makes error handling more
+predictable, since our catch block will not receive errors from any background
+task, we're better able to specify which errors we actually want to deal with.
+
+## Error boundary
+
+If we do want to catch an error from a spawned task (or from a [Resource][]) then
+we need to introduce an intermediate task which allows us to bring the error into
+the foreground. We call this pattern an "error boundary":
+
+``` typescript
+import { main } from 'effection';
+import { tickingBomb } from './ticking-bomb';
+
+main(function*() {
+  try {
+    yield function*() { // error boundary
+      yield spawn(tickingBomb()); // will blow up in the background
+      yield; // sleep forever
+    }
+  } catch(err) {
+    console.log("it blew up:", err.message);
+  }
+});
+```
+
+## Main error
+
+We have used the `main` entry point to run our code to run our code, when we build
+our entire application using Effection. One advantage of `main` over `run` is that
+when our operation fails, we exit the program with a proper failure exit code.
+
+However, there are cases where we want the program to exit, but it might be due
+to user error, rather than an internal failure. In this case we might not want
+to print a stack trace. For example, if we're building a CLI which reads a
+file, and the file that the user has specified does not exist, then we might want to
+show a message to the user, but there is no need to show a stack trace. Additonally,
+we might want to set a specific exit code.
+
+Effection ships with a special error type, `MainError`, which works together
+with `main` for these types of situations:
+
+``` typescript
+import { main, MainError } from 'effection';
+import { promises as fs } from 'fs';
+
+const { readFile } = fs;
+
+main(function*() {
+  let fileName = process.argv[2];
+  try {
+    let content = yield readFile(fileName);
+    console.log(content.reverse().toString());
+  } catch(err) {
+    throw new MainError({ message: `no such file ${fileName}`, exitCode: 200 });
+  }
+});
+```
+
+When using `MainError` the stack trace will not be printed and the exit code
+specified in the error will be used.
```

Comment:
I think an example of the difference will really cement the concept. What about showing this example code running both with and without a `MainError`